client/client.go

```diff
@@ -0,0 +1,263 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	notion "github.com/conduitio-labs/notionapi"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+var (
+	ErrPageNotFound = errors.New("page not found")
+)
+
+type Page struct {
+	ID string
+	// Parent is a JSON string describing the page's parent object
+	Parent string
+	URL    string
+	// CreatedBy is a JSON string describing the user who created this page
+	CreatedBy   string
+	CreatedTime time.Time
+	// LastEditedBy is a JSON string describing the user who last edited this page
+	LastEditedBy   string
+	LastEditedTime time.Time
+	Archived       bool
+
+	properties notion.Properties
+	children   []notion.Block
+}
+
+func NewPage(pg *notion.Page, children []notion.Block) Page {
+	return Page{
+		ID:             pg.ID.String(),
+		Parent:         toJSON(pg.Parent),
+		URL:            pg.URL,
+		CreatedTime:    pg.CreatedTime,
+		CreatedBy:      toJSON(pg.CreatedBy),
+		LastEditedBy:   toJSON(pg.LastEditedBy),
+		LastEditedTime: pg.LastEditedTime,
+		Archived:       pg.Archived,
+		properties:     pg.Properties,
+		children:       children,
+	}
+}
+
+// PlainText returns a plain text representation of a page
+func (p Page) PlainText(ctx context.Context) (string, error) {
+	var plainText string
+	for _, c := range p.children {
+		text, err := extractText(c)
+		if errors.Is(err, errNoExtractor) {
+			sdk.Logger(ctx).Warn().
+				Str("block_type", c.GetType().String()).
+				Msg("no text extractor registered")
+			continue
+		}
+		if err != nil {
+			return "", err
+		}
+		plainText += text + "\n"
+	}
+
+	return plainText, nil
+}
+
+// Title returns a page's title.
+func (p Page) Title() string {
+	if len(p.properties) == 0 {
+		return ""
+	}
+
+	tp, ok := p.properties["title"].(*notion.TitleProperty)
+	if !ok || len(tp.Title) == 0 {
+		return ""
+	}
+
+	return tp.Title[0].PlainText
+}
+
+type DefaultClient struct {
```

Comment:
Food for thought - I think we can find a better name than `DefaultClient`. I'm not sure if there's a better word than "Client" to differentiate it a bit from `notion.Client`. Also `client.DefaultClient` stutters, and given that there's no other client in this package "default" is kinda redundant.

My recommendation would be to do 2 things:
- rename the package to `source`, this would give it a straightforward purpose, it contains utilities needed for the source connector (which should stay in the root, to be clear)
- rename `DefaultClient` to just `Client` (or `Fetcher`, to avoid the word "client", but I'm not sure if that's better)

Then we have `source.Client` (or `source.Fetcher`).

Reply:
Good thoughts, I like them. I initially put all the client code into a separate package, because I thought we'll have a Notion destination soon-ish, so some code can be re-used. But there are no plans like that for the near future, so I think `source.Client` makes sense. :+1: (And I like `Client` more then `Fetcher`.)